tests/documents_tests.ts

```diff
@@ -37,582 +34,631 @@ afterAll(() => {
-        'code',
-        ErrorStatusCode.INVALID_API_KEY
-      )
-    })
-
-    test(`${permission} key: Try to get documents and be denied`, async () => {
-      await expect(client.getIndexes()).rejects.toHaveProperty(
-        'code',
-        ErrorStatusCode.INVALID_API_KEY
-      )
-    })
-
-    test(`${permission} key: Try to delete one document and be denied`, async () => {
-      await expect(client.getIndexes()).rejects.toHaveProperty(
-        'code',
-        ErrorStatusCode.INVALID_API_KEY
-      )
-    })
-
-    test(`${permission} key: Try to delete some documents and be denied`, async () => {
-      await expect(client.getIndexes()).rejects.toHaveProperty(
-        'code',
-        ErrorStatusCode.INVALID_API_KEY
-      )
-    })
-
-    test(`${permission} key: Try to delete all documents and be denied`, async () => {
-      await expect(client.getIndexes()).rejects.toHaveProperty(
-        'code',
-        ErrorStatusCode.INVALID_API_KEY
+  )
+
+  describe.each([{ permission: 'No' }])(
+    'Test on documents',
+    ({ permission }) => {
+      beforeEach(() => {
+        return clearAllIndexes(config)
+      })
+
+      test(`${permission} key: Try to add documents and be denied`, async () => {
+        const client = await getClient(permission)
+
+        await expect(client.getIndexes()).rejects.toHaveProperty(
+          'code',
+          ErrorStatusCode.MISSING_AUTHORIZATION_HEADER
+        )
+      })
+
+      test(`${permission} key: Try to update documents and be denied`, async () => {
+        const client = await getClient(permission)
+        await expect(client.getIndexes()).rejects.toHaveProperty(
+          'code',
+          ErrorStatusCode.MISSING_AUTHORIZATION_HEADER
+        )
+      })
+
+      test(`${permission} key: Try to get documents and be denied`, async () => {
+        const client = await getClient(permission)
+        await expect(client.getIndexes()).rejects.toHaveProperty(
+          'code',
+          ErrorStatusCode.MISSING_AUTHORIZATION_HEADER
+        )
+      })
+
+      test(`${permission} key: Try to delete one document and be denied`, async () => {
+        const client = await getClient(permission)
+        await expect(client.getIndexes()).rejects.toHaveProperty(
+          'code',
+          ErrorStatusCode.MISSING_AUTHORIZATION_HEADER
+        )
+      })
+
+      test(`${permission} key: Try to delete some documents and be denied`, async () => {
+        const client = await getClient(permission)
+        await expect(client.getIndexes()).rejects.toHaveProperty(
+          'code',
+          ErrorStatusCode.MISSING_AUTHORIZATION_HEADER
+        )
+      })
+
+      test(`${permission} key: Try to delete all documents and be denied`, async () => {
+        const client = await getClient(permission)
+        await expect(client.getIndexes()).rejects.toHaveProperty(
+          'code',
+          ErrorStatusCode.MISSING_AUTHORIZATION_HEADER
+        )
+      })
+    }
+  )
```

Comment:
I still don't see the differences between all these tests. Even if they add differents names is the same pattern in all tests:
```
        const client = await getClient(permission)
        await expect(client.getIndexes()).rejects.toHaveProperty(
          'code',
          ErrorStatusCode.MISSING_AUTHORIZATION_HEADER
        )
```